deflate.h

```diff
@@ -106,26 +106,35 @@ typedef uint16_t Pos;
 
 typedef struct internal_state {
     PREFIX3(stream)      *strm;            /* pointer back to this zlib stream */
-    int                  status;           /* as the name implies */
     unsigned char        *pending_buf;     /* output still pending */
     unsigned long        pending_buf_size; /* size of pending_buf */
     unsigned char        *pending_out;     /* next pending byte to output to the stream */
     uint32_t             pending;          /* nb of bytes in the pending buffer */
     int                  wrap;             /* bit 0 true for zlib, bit 1 true for gzip */
     PREFIX(gz_headerp)   gzhead;           /* gzip header information to write */
     uint32_t             gzindex;          /* where in extra, name, or comment */
+    int                  status;           /* as the name implies */
     unsigned char        method;           /* can only be DEFLATED */
```

Comment:
We should avoid stray char/unsigned char in structures... In worst case it will introduce 3-7 byte padding. Accessing single byte is usually very slow compared to accessing 4 bytes.

Reply:
I agree, but that one is not touched in this PR.
I have a few more changes I want to make, but in order to make the PR reviewable, bisectable and testable, I decided to split it up into bite-sized pieces.

I am considering making `status` and `method` uint16_t instead, but have not tested yet. (Well, I did, but as part of a much bigger changeset, and that turned out to be unreviewable, untestable and unbisectable.)
Compared to 32-bits, reads of 16-bits are not bad, but writes take a little hit. 8-bits however take a big write hit and a bigger read hit.
With them 16-bits, they also fit in 32-bits together and remove a padding hole of 3 bytes (In total freeing up 4 bytes of the struct).
Will have to test to see whether it is a win or a loss, and whether it makes a mess of conversion trouble in other places.